PracticeProject/ViewModel/UserViewModel.swift

```diff
@@ -0,0 +1,42 @@
+//
+//  UserViewModel.swift
+//  PracticeProject
+//
+//  Created by Gnanavel on 31/03/22.
+//
+
+import Foundation
+import Combine
+
+
+/// UserViewModel is handling the user selection functionality like api and other business logics
+class UserViewModel: ObservableObject {
+    @Published var userList = [UserModel]()
+    @Published var loading = false
+    var errorMessage: String = ""
+
+    let request: API
+    init(request: API = UserAPI()) {
+        self.request = request
+    }
+    
+    /// Getting the user list from API response
+    /// - Parameter excludingUserWithID: It is excluding the selected user
+    func loadData(excludingUserWithID: String) {
+        self.loading = true
+        request.fetchUsers(excludingUserWithID: excludingUserWithID) { userList in
+            self.loading = false
```

Comment:
There is a strong reference to `self` in this async callback block. What are the dangers of that? How can you fix it? 
(Elaborate)

Reply:
@nets-dev Yes, you are right, Forgot to add the weak reference in this async method. So, I have changed by using `[weak self]` to fix it.

Reply:
* What is the benefit of `weak self` here? 
* How does it compare to using `self` with strong reference? 
* How does it compare to using `unowned self`